internal/sftp/goclient_test.go

```diff
@@ -287,4 +286,60 @@ func TestGoClient(t *testing.T) {
 			assert.Assert(t, tfs.Equal(dest.Path(), tfs.Expected(t, tc.want.Paths...)))
 		})
 	}
+
+	type deleteTest struct {
+		name     string
+		cfg      sftp.Config
+		bytes    int64
+		filename string
+		wantErr  string
+	}
+	for _, tc := range []deleteTest{
+		{
+			name: "Deletes a file",
+			cfg: sftp.Config{
+				Host:           host,
+				Port:           port,
+				KnownHostsFile: "./testdata/known_hosts",
+				PrivateKey: sftp.PrivateKey{
+					Path: "./testdata/clientkeys/test_ed25519",
+				},
+			},
+			bytes:    13,
+			filename: "test.txt",
+		},
+		{
+			name: "Errors when file doesn't exist",
+			cfg: sftp.Config{
+				Host:           host,
+				Port:           port,
+				KnownHostsFile: "./testdata/known_hosts",
+				PrivateKey: sftp.PrivateKey{
+					Path: "./testdata/clientkeys/test_ed25519",
+				},
+			},
+			filename: "error.txt",
+			wantErr:  "SFTP: unable to get file info for",
+		},
+	} {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			sftpc := sftp.NewGoClient(tc.cfg)
+			src := strings.NewReader("Testing 1-2-3")
+			dest := tfs.NewDir(t, "sftp_test")
+
+			sftpc.Upload(context.Background(), src, dest.Join("test.txt"))
```

Comment:
This test will fail if the `Upload()` method fails, which isn't ideal.  Because the SFTP server is using the same filesystem as the client, you can create a file directly in `dest` then call `sftpc.Delete()` to delete it.

Reply:
Yes this is a better idea.